### ISSUE TITLE
Celery task fails because of custom Django user model

### ISSUE THREAD

**Issue by kviktor** [celery/celery#3048]
Whenever I call a task on its queue I get this error. The WARNING/PoolWorker-1 message is `print(settings.INSTALLED_APPS)` (so the profile app is in INSTALLED apps, I don't know why celery can't find it). I'm using the current Celery master. (somewhat related issue #3041)

```
 -------------- celery@hostname v4.0.0rc1 (0today8)
---- **** -----
--- * ***  * -- Linux-3.13.0-48-generic-x86_64-with-Ubuntu-14.04-trusty 2016-02-10 14:52:07
-- * - **** ---
- ** ---------- [config]
- ** ---------- .> app:         manager:0x7fedb4977278
- ** ---------- .> transport:   amqp://user:**@127.0.0.1:5672/project
- ** ---------- .> results:     amqp
- *** --- * --- .> concurrency: 1 (prefork)
-- ******* ----
--- ***** ----- [queues]
 -------------- .> manager          exchange=manager(direct) key=manager
               
 
[tasks]
  . belam
  . execcelery.gcc_compile
  . execcelery.run_program
  . execcelery.run_script
 
[2016-02-10 14:52:08,026: INFO/MainProcess] Connected to amqp://user:**@127.0.0.1:5672/project
[2016-02-10 14:52:08,041: INFO/MainProcess] mingle: searching for neighbors
[2016-02-10 14:52:09,067: INFO/MainProcess] mingle: sync with 1 nodes
[2016-02-10 14:52:09,068: INFO/MainProcess] mingle: sync complete
celery/fixups/django.py:300: UserWarning: Using s
ettings.DEBUG leads to a memory leak, never use this setting in production environments!
  warnings.warn('Using settings.DEBUG leads to a memory leak, never '
 
[2016-02-10 14:52:09,082: WARNING/MainProcess] ce
lery/fixups/django.py:300: UserWarning: Using settings.DEBUG leads to a memory leak, never use this settin
g in production environments!
  warnings.warn('Using settings.DEBUG leads to a memory leak, never '
 
[2016-02-10 14:52:09,084: WARNING/MainProcess] celery@hostname ready.
0[2016-02-10 14:52:53,147: INFO/MainProcess] Received task: belam[f230b09e-34c2-4d14-a365-16ff8cc3e591]  
[2016-02-10 14:52:53,753: WARNING/PoolWorker-1] ('django.contrib.auth', 'django.contrib.contenttypes', 'dj
ango.contrib.sessions', 'django.contrib.sites', 'django.contrib.messages', 'django.contrib.staticfiles', '
django.contrib.admin', 'django_tables2', 'crispy_forms', 'taggit', 'acl', 'widget_tweaks', 'sizefield', 'a
utocomplete_light', 'profile', 'course', 'lecture', 'grading', 'exercise', 'assignment', 'dashboard', 'man
agement', 'rosetta')
[2016-02-10 14:52:53,801: ERROR/PoolWorker-1] Task belam[f230b09e-34c2-4d14-a365-16ff8cc3e591] raised unex
pected: ValueError("Related model 'profile.ProjectUser' cannot be resolved",)
Traceback (most recent call last):
  File "celery/app/trace.py", line 358, in trace_
task
    R = retval = fun(*args, **kwargs)
  File "celery/app/trace.py", line 609, in __prot
ected_call__
    return self.run(*args, **kwargs)
  File "project/project/exercise/tasks.py", line 9, in evaluate_submission
    Submission.objects.get(pk=submission_pk).evaluate()
  File "django/db/models/manager.py", line 127, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "django/db/models/query.py", line 328, in
get
    num = len(clone)
  File "django/db/models/query.py", line 144, in
__len__
    self._fetch_all()
  File "django/db/models/query.py", line 965, in
_fetch_all
    self._result_cache = list(self.iterator())
  File "django/db/models/query.py", line 238, in
iterator
    results = compiler.execute_sql()
  File "django/db/models/sql/compiler.py", line 8
29, in execute_sql
    sql, params = self.as_sql()
  File "django/db/models/sql/compiler.py", line 3
78, in as_sql
    extra_select, order_by, group_by = self.pre_sql_setup()
  File "django/db/models/sql/compiler.py", line 4
8, in pre_sql_setup
    self.setup_query()
  File "django/db/models/sql/compiler.py", line 3
9, in setup_query
    self.select, self.klass_info, self.annotation_col_map = self.get_select()
  File "django/db/models/sql/compiler.py", line 1
87, in get_select
    for c in self.get_default_columns():
  File "django/db/models/sql/compiler.py", line 5
24, in get_default_columns
    column = field.get_col(alias)
  File "django/db/models/fields/related.py", line
 2027, in get_col
    return super(ForeignKey, self).get_col(alias, output_field or self.related_field)
  File "django/db/models/fields/related.py", line
 1907, in related_field
    return self.foreign_related_fields[0]
  File "django/db/models/fields/related.py", line
 1641, in foreign_related_fields
    return tuple(rhs_field for lhs_field, rhs_field in self.related_fields if rhs_field)
  File "django/db/models/fields/related.py", line
 1628, in related_fields
    self._related_fields = self.resolve_related_fields()
  File "django/db/models/fields/related.py", line
 1613, in resolve_related_fields
    raise ValueError('Related model %r cannot be resolved' % self.rel.to)
ValueError: Related model 'profile.ProjectUser' cannot be resolved
```

edit:
Added a print to line 1613 in `django/db/models/fields/related.py`
```                                                      
print(self.rel)                                                           
print(self.rel.to)                                                        
```

When celery starts I get these (related) messages
```
<ManyToOneRel: exercise.submission>
<class 'assignment.models.Assignment'>
```

However when I start up the webserver (`./manage.py runserver`) I get the missing relation
```
<ManyToOneRel: exercise.submission>
<class 'assignment.models.Assignment'>
<ManyToOneRel: exercise.submission>
<class 'profile.models.ProjectUser'>
```

**Comment by kviktor** [comment]
I think I found the problem, it lies within the django module loading of celery.


Our custom user model lives in a module called profile and that seems to be the problem.

I added this print `print(vars(self.app_configs['profile']))` to `django.apps.registry.Apps.populate` and got this result when starting celery (celery -A project worker)
```
{'models': None, 'verbose_name': 'Profile', 'label': 'profile', 'models_module': None, 'name': 'profile', 'path': '/usr/lib/python3.4', 'module': <module 'profile' from '/usr/lib/python3.4/profile.py'>}
```

However when I start django's webserver (`./manage.py runserver`) I get a KeyError: 'profile'. So it looks like Django is loading the modules in another way. After renaming the module everything looks/working fine.
